### PR TITLE
chore(api): add async to create/update/delete operations in GraphQLConectionScenario(1,2)Tests

### PR DESCRIPTION
**Pull request by manaswi223**
*Issue #, if available:*

*Description of changes:*
Changes made to migrate old APIs to new async APIs on create/update/delete operations for `GraphQLConectionScenario1Tests` and `GraphQLConectionScenario2Tests` .
Also, added async API changes to Create/update operations for PostSubscriptionWithModels in `GraphQLModelBasedTests`

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lawmicha** [comment]
since these are integration test changes, the title should be `chore`. we usually use `fix` for bug fixes and `feat` for new public API changes.

File: .swiftpm/xcode/xcshareddata/xcschemes/AWSAPIPlugin.xcscheme [hunk 1]
```diff
@@ -20,6 +20,20 @@
...
+            buildForArchiving = "NO"
+            buildForAnalyzing = "NO">
+            <BuildableReference
+               BuildableIdentifier = "primary"
+               BlueprintIdentifier = "AWSAPIPluginTests"
+               BuildableName = "AWSAPIPluginTests"
+               BlueprintName = "AWSAPIPluginTests"
+               ReferencedContainer = "container:">
+            </BuildableReference>
+         </BuildActionEntry>
```
Comment (lawmicha) [inline .swiftpm/xcode/xcshareddata/xcschemes/AWSAPIPlugin.xcscheme:36]: what is this for? do we need it?

File: Package.swift [hunk 2]
```diff
@@ -16,7 +16,7 @@ let dependencies: [Package.Dependency] = [
     .package(
         name: "AWSSwiftSDK",
         url: "https://github.com/awslabs/aws-sdk-swift.git",
-        .upToNextMinor(from: "0.2.6")
+        .upToNextMinor(from: "0.2.5")
```
Comment (lawmicha) [inline Package.swift:19]: can you revert this change?

File: AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario2Tests.swift [hunk 3]
```diff
@@ -54,137 +54,90 @@ class GraphQLConnectionScenario2Tests: XCTestCase {
...
+        _ = try await Amplify.API.mutate(request: .create(team2))
         let project2a = Project2(teamID: team2.id, team: team2)
-        Amplify.API.mutate(request: .create(project2a)) { result in
-            switch result {
-            case .success:
-                createProject2aSuccessful.fulfill()
-            case .failure(let error):
-                XCTFail("\(error)")
+        _ = try await Amplify.API.mutate(request:
+            .create(project2a))
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario2Tests.swift:62]: place on same line

File: AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario2Tests.swift [hunk 4]
```diff
@@ -54,137 +54,90 @@ class GraphQLConnectionScenario2Tests: XCTestCase {
...
-            case .failure(let error):
-                XCTFail("\(error)")
+        
+        let project2b = Project2(teamID: "randomTeamID", team: team2)
+        _ = try await Amplify.API.mutate(request: .create(project2b))
+        let result2 = try await Amplify.API.query(request: .get(Project2.self, byId: project2b.id))
+        switch result2 {
+        case .success(let queriedProjectOptional):
+            guard let queriedProject = queriedProjectOptional else {
+                XCTFail("Failed")
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario2Tests.swift:83]: "querying for the project should not return nil"

File: AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario2Tests.swift [hunk 5]
```diff
@@ -54,137 +54,90 @@ class GraphQLConnectionScenario2Tests: XCTestCase {
...
+                return
+            }
+            createdProject.team = createdAnotherTeam
+    
+            guard case .success(let updatedProject) = try await Amplify.API.mutate(request: .update(createdProject)) else {
+                XCTFail("Failed to update project to another team")
+                return
             }
+            XCTAssertEqual(updatedProject.teamID, anotherTeam.id)
+//          XCTAssertEqual(updatedProject.team, anotherTeam)
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLConnectionScenario2Tests.swift:112]: let's add a comment, indicating why we have commented this out like
```
// The team object does not get retrieved from the service and is `nil`, but should be eager loaded to contain the `team`
XCTAssertEqual(updatedProject.team, anotherTeam)
```

File: Package.swift (hunk 2 above)
Comment (manaswi223) [reply Package.swift]: done

File: .swiftpm/xcode/xcshareddata/xcschemes/AWSAPIPlugin.xcscheme (hunk 1 above)
Comment (manaswi223) [reply .swiftpm/xcode/xcshareddata/xcschemes/AWSAPIPlugin.xcscheme:36]: I think these are the schema changes when I move the target to `AWSAPIPluginTests`, shall I discard these changes ?

Comment (lawmicha) [reply .swiftpm/xcode/xcshareddata/xcschemes/AWSAPIPlugin.xcscheme:36]: This looks like it is under the AWSAPIPlugin scheme, which should build the AWSAPIPlugin target with CMD+B and execute the tests using AWSAPIPluginTests with CMD+U. Looks like AWSAPIPluginTests is added and set to "build for running" which looks inaccurate. I would discard this change and see if you need anything else to achieve the development workflow of building and testing AWSAPIPlugin, which i think was already set up.

Comment (manaswi223) [reply .swiftpm/xcode/xcshareddata/xcschemes/AWSAPIPlugin.xcscheme]: oh sure, discarded the changes

**Review by lawmicha — APPROVED** [review]
